CMakeLists.txt

```
@@ -90,16 +92,49 @@ if(NEOFOAM_BUILD_DOC)
   neofoam_build_docs()
 endif()
 
-add_custom_target(
-  cmake-format
-  COMMAND find . -name CMakeLists.txt -exec cmake-format -i {} +
-  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
-
-add_custom_target(
-  format
-  COMMAND find  include src applications test benchmarks -name "*.[c,h]*pp" -exec
-          clang-format -i {} +
-  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
+if(NEOFOAM_DEVEL_TOOLS)
+  find_program(PRE_COMMIT pre-commit)
+  if(NOT PRE_COMMIT)
+    message(
+      FATAL_ERROR
+        "The pre-commit command was not found."
+        "It is necessary if you want to commit changes to NeoFOAM."
+        "If that is not the case, set NEOFOAM_DEVEL_TOOLS=OFF. "
+        "Otherwise install pre-commit via pipx (or pip) using:\n"
+        "    pipx install pre-commit")
+  endif()
+
+  execute_process(
+    COMMAND "${PRE_COMMIT}" "install"
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    RESULT_VARIABLE pre-commit-result
+    OUTPUT_VARIABLE pre-commit-output
+    ERROR_VARIABLE pre-commit-error)
+  if(pre-commit-result)
+    message(
+      FATAL_ERROR
+        "Failed to install the git hooks via pre-commit."
+        "Please check the error message:\n"
+        "${pre-commit-output}\n${pre-commit-error}")
+  endif()
+  if(pre-commit-output MATCHES "^Running in migration mode with existing hooks")
+    message(
+      WARNING
+        "An existing git hook was encountered during `pre-commit install`."
+        "The old git hook will also be executed."
+        "Consider removing it with `pre-commit install -f`")
+  elseif(NOT pre-commit-output MATCHES "^pre-commit installed at")
+    message(WARNING "`pre-commit install` did not exit normally."
+                    "Please check the output message:\n" "${pre-commit-output}")
+  endif()
+
+  add_custom_target(
+    format
```

```suggestion
    pre-commit
```
might be more descriptive